src/Provider/IslandoraServiceProvider.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Islandora\Crayfish\Commons\Provider;
+
+use Islandora\Crayfish\Commons\CmdExecuteService;
+use Islandora\Crayfish\Commons\FedoraResourceConverter;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
+use Silex\Provider\DoctrineServiceProvider;
+use Silex\Provider\MonologServiceProvider;
+use Silex\Provider\ServiceControllerServiceProvider;
+use Silex\Provider\SecurityServiceProvider;
+use Islandora\Chullo\FedoraApi;
+use Islandora\Crayfish\Commons\Syn\SettingsParser;
+use Islandora\Crayfish\Commons\Syn\JwtAuthenticator;
+use Islandora\Crayfish\Commons\Syn\JwtFactory;
+
+class IslandoraServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register(Container $container)
+    {
+        // Register services we rely on
+        $container->register(new MonologServiceProvider());
```

Comment:
I disagree on those points Jonathan but again, I won't oppose, it is just not the silex way. Testing services that don't register their own dependencies is in fact easier. I get your point that this may not a general use service, ok, but imagine the case that another service (islandora one) requires access to database under different means or another related services uses monolog under a different configuration. By obscuring the dependencies in the code, an implementer could override (in a bad for us, but in an silex expected way) what you already build here at service provider (settings etc) and since the whole silex/pimple idea is based on lazy loading, means nothing is initialized, set, called until it is used it is complicated to define upfront which of the registrations will in fact override the other one.